azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -339,30 +339,41 @@ protected Azure getOrCreateAzureClient() throws AzureAuthFailureException, Azure
             mavenAuthConfiguration.setType(getAuthType());
 
             SystemPropertyUtils.injectCommandLineParameter("auth", mavenAuthConfiguration, MavenAuthConfiguration.class);
-            azureCredentialWrapper = MavenAuthManager.getInstance().login(session, settingsDecrypter, mavenAuthConfiguration);
-
-            if (Objects.isNull(azureCredentialWrapper)) {
+            MavenAuthManager.getInstance().login(MavenAuthManager.getInstance().buildAuthConfiguration(session, settingsDecrypter, mavenAuthConfiguration));
+            Account account = com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).account();
+            if (!account.isAuthenticated()) {
                 return null;
             }
-
-            final AzureEnvironment env = azureCredentialWrapper.getEnv();
+            final AzureEnvironment env = account.getEnvironment();
             final String environmentName = AzureEnvironmentUtils.azureEnvironmentToString(env);
             if (env != AzureEnvironment.AZURE) {
                 Log.prompt(String.format(USING_AZURE_ENVIRONMENT, TextUtils.cyan(environmentName)));
             }
-            Log.info(azureCredentialWrapper.getCredentialDescription());
-            final PagedList<Subscription> subscriptions = Azure.configure()
-                    .authenticate(azureCredentialWrapper.getAzureTokenCredentials()).subscriptions().list();
-            subscriptions.loadAll();
-            final String targetSubscriptionId = getTargetSubscriptionId(getSubscriptionId(), subscriptions);
+            printCredentialDescription(account);
+            final List<SubscriptionEntity> subscriptions = account.getSubscriptions();
+            final String targetSubscriptionId = getTargetSubscriptionId(getSubscriptionId(), subscriptions, account.getSelectedSubscriptions());
             checkSubscription(subscriptions, targetSubscriptionId);
-            azureCredentialWrapper.withDefaultSubscriptionId(targetSubscriptionId);
-            return AzureClientFactory.getAzureClient(azureCredentialWrapper, getUserAgent());
+
+            this.subscriptionId = targetSubscriptionId;
+            return AzureClientFactory.getAzureClient(getUserAgent(), targetSubscriptionId);
         } catch (AzureLoginException | IOException e) {
             throw new AzureAuthFailureException(e.getMessage());
         }
     }
 
+    protected void printCredentialDescription(Account account) {
```

Comment:
Move to Account library to reduceduplicate codes?

Reply:
Good idea

Reply:
The only concern is here I need a common ILog or Messager for prompting something, since log levels in maven and gradle are different